main/webapp/modules/core/scripts/project/operation-icons.js

```diff
@@ -0,0 +1,13 @@
+
+var OperationIconRegistry = {
+  map: new Map()
+};
+
+OperationIconRegistry.setIcon = function(operationId, iconPath) {
+  this.map.set(operationId, iconPath);
```

Comment:
```suggestion
  // this will cause the image to load so that once it is added to the DOM it will already be in the browser cache
  var image = new Image();
  image.src = iconPath;

  this.map.set(operationId, iconPath);
```